core/templates/pages/exploration-player-page/services/exploration-recommendations.service.spec.ts

```diff
@@ -0,0 +1,155 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ExplorationRecommendationsService.
+ */
+
+
+import { async, TestBed } from '@angular/core/testing';
+import { ExplorationRecommendationsService } from
+  // eslint-disable-next-line max-len
+  'pages/exploration-player-page/services/exploration-recommendations.service';
+import { ContextService } from 'services/context.service';
+import { ServicesConstants } from 'services/services.constants';
+import { UrlService } from 'services/contextual/url.service';
+import { ExplorationRecommendationsBackendApiService } from
+  'domain/recommendations/exploration-recommendations-backend-api.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+
+
+
+describe('Exploration Recommendations Service', () => {
+  let expRecsService: ExplorationRecommendationsService;
+  let urlService: UrlService;
+  let contextService: ContextService;
+  let expRecsBackendApiService: ExplorationRecommendationsBackendApiService;
+  const STORY_ID = '1';
+  const COLLECTION_ID = '2';
+  const NODE_ID = '3';
+  const EXPLORATION_ID = '4';
+  const AUTHOR_REC_EXP_IDS = ['5', '6'];
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+      ],
+      providers: [
+        ExplorationRecommendationsService,
+      ]
+    });
+  }));
+
+  beforeEach(() => {
+    urlService = TestBed.get(UrlService);
+    contextService = TestBed.get(ContextService);
+    expRecsBackendApiService = TestBed.get(
+      ExplorationRecommendationsBackendApiService
+    );
+
+    spyOn(urlService, 'getCollectionIdFromExplorationUrl').and.returnValue(
+      COLLECTION_ID);
+
+    spyOn(urlService, 'getUrlParams').and.returnValue({
+      story_id: STORY_ID,
+      node_id: NODE_ID
+    });
+
+    spyOn(contextService, 'getExplorationId').and.returnValue(EXPLORATION_ID);
+
+    spyOn(
+      expRecsBackendApiService, 'getRecommendedSummaryDicts').and.resolveTo([]);
+  });
+
+  describe('when context is in editor page', () => {
+    beforeEach(() => {
+      spyOn(contextService, 'getPageContext').and.returnValue(
+        ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR
+      );
+    });
+
+    it('should initialize with correct context', () => {
+      expRecsService = TestBed.get(ExplorationRecommendationsService);
+      expect(expRecsService.isInEditorPage).toBeTrue();
+    });
+
+    describe('in preview mode', () => {
+      beforeEach(() => {
+        spyOn(contextService, 'getEditorTabContext').and.returnValue(
+          ServicesConstants.EXPLORATION_EDITOR_TAB_CONTEXT.PREVIEW
+        );
+      });
+      it('should initialize with correct context', () => {
+        expRecsService = TestBed.get(ExplorationRecommendationsService);
+        expect(expRecsService.isInEditorPage).toBeTrue();
+        expect(expRecsService.isInEditorPreviewMode).toBeTrue();
+      });
+    });
+    describe('getRecommendendedSummaryDicts', () => {
```

Comment:
Please add a newline above this one, for readability.

Reply:
@TheoLipeles please make sure you respond to all comments on the PR. Thanks!

Reply:
no longer relevant, this section was moved